tini/src/seccomp_fd_notify.c

```diff
@@ -179,23 +184,60 @@ struct sock_filter net_filter[] = {
 		}
 	}
 
+	for (int i = 0;
+	     i < (unsigned short)(sizeof(check_filter) / sizeof(check_filter[0])) &&
+		 filter_idx < BPF_MAXINSNS;
+		 i++) {
+		filter[filter_idx] = check_filter[i];
+		filter_idx++;
+	}
+
 	char *is_handle_net = getenv("TITUS_SECCOMP_AGENT_HANDLE_NET_SYSCALLS");
 	char *is_handle_perf = getenv("TITUS_SECCOMP_AGENT_HANDLE_PERF_SYSCALLS");
+	char *is_traffic_steering = getenv("TITUS_SECCOMP_AGENT_HANDLE_TRAFFIC_STEERING");
 	if (is_handle_net != NULL) {
-		prog.filter = net_filter;
-		prog.len = 
-			(unsigned short)(sizeof(net_filter) / sizeof(net_filter[0]));
+		for (int i = 0;
+		     i < (unsigned short)(sizeof(net_filter) / sizeof(net_filter[0])) &&
+			 filter_idx < BPF_MAXINSNS;
+		     i++) {
+			filter[filter_idx] = net_filter[i];
+			filter_idx++;
+		}
 		PRINT_INFO("Networking system calls will be intercepted");
 	} else if (is_handle_perf != NULL) {
-		prog.filter = perf_filter;
-		prog.len = 
-			(unsigned short)(sizeof(perf_filter) / sizeof(perf_filter[0]));
+		for (int i = 0;
+		     i < (unsigned short)(sizeof(perf_filter) / sizeof(perf_filter[0])) &&
+			 filter_idx < BPF_MAXINSNS;
+			 i++) {
+			filter[filter_idx] = perf_filter[i];
+			filter_idx++;
+		}
 		PRINT_INFO("BPF/Perf system calls will be intercepted");
+	} else if (is_traffic_steering != NULL) {
+		for (int i = 0;
+			 i < (unsigned short)(sizeof(traffic_steering_filter) / sizeof(traffic_steering_filter[0])) &&
+			 filter_idx < BPF_MAXINSNS;
+			 i++) {
+			filter[filter_idx] = traffic_steering_filter[i];
+			filter_idx++;
+		}
+		PRINT_INFO("traffic steering system calls will be intercepted");
 	} else {
 		PRINT_INFO("No env variables set, no interception of system calls");
 		return -1;
 	}
 
+	for (int i = 0;
+	     i < (unsigned short)(sizeof(allow_filter) / sizeof(allow_filter[0])) &&
+	     filter_idx < BPF_MAXINSNS;
+		 i++) {
+		filter[filter_idx] = allow_filter[i];
+		filter_idx++;
+	}
+
+	prog.filter = filter;
+	prog.len = filter_idx;
```

Comment:
Whoa, is this right? Are we nesting the filters now instead of concatenating them?

Reply:
Not nesting, we are still concatenating based on what config is enabled.